crates/extensions/tedge_timer_ext/src/builder.rs

```diff
@@ -49,50 +47,19 @@ impl RuntimeRequestSink for TimerActorBuilder {
 }
 
 impl<T: Message> ServiceProvider<SetTimeout<T>, Timeout<T>, NoConfig> for TimerActorBuilder {
-    fn add_peer(&mut self, peer: &mut impl ServiceConsumer<SetTimeout<T>, Timeout<T>, NoConfig>) {
-        let mut adapter = AnyTimerAdapter::new(peer);
-        self.box_builder.add_peer(&mut adapter);
-    }
-}
-
-/// A message adapter used by actors to send timer requests with a generic payload `SetTimeout<T>`
```

Comment:
To be honest, I didn't really understand how the new `connect_consumer` API enabled this simplification. I'm struggling to understand that connection.

Reply:
The former API was forcing the introduction of some concrete type implementing the `ServiceConsumer` trait. Now you can connect a `ServiceProducer` without such an implementation.